src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageItemListViewModel.cs

```diff
@@ -369,7 +394,69 @@ public bool IsPackageDeprecated
             }
         }
 
-        public Uri IconUrl { get; set; }
+        private Uri _iconUrl;
+        public Uri IconUrl
+        {
+            get { return _iconUrl; }
+            set
+            {
+                _iconUrl = value;
+                OnPropertyChanged(nameof(IconUrl));
+            }
+        }
+
+        private IconBitmapStatus _bitmapStatus;
+
+        public IconBitmapStatus BitmapStatus
+        {
+            get { return _bitmapStatus; }
+            set
+            {
+                if (_bitmapStatus != value)
+                {
+                    _bitmapStatus = value;
+                    OnPropertyChanged(nameof(BitmapStatus));
+                }
+            }
+        }
+
+        private BitmapSource _iconBitmap;
+        public BitmapSource IconBitmap
+        {
+            get
+            {
+                if (_iconBitmap == null)
+                {
+                    if (BitmapStatus == IconBitmapStatus.None)
+                    {
+                        (BitmapSource iconBitmap, IconBitmapStatus nextStatus) = GetInitialIconBitmapAndStatus();
+
+                        BitmapStatus = nextStatus;
+                        if (BitmapStatus == IconBitmapStatus.NeedToFetch)
+                        {
+                            BitmapStatus = IconBitmapStatus.Fetching;
```

Comment:
testing before i merge found that my default icon while fetching was broken during one of changes a day or two ago.
fixing that ... and adding a test to cover that case.

Reply:
fixed in current iteration.